proxy/StatPages.cc

```diff
@@ -83,7 +83,7 @@ StatPagesManager::handle_http(Continuation *cont, HTTPHdr *header)
     host_len       = unescapifyStr(host);
 
     for (i = 0; i < n_stat_pages; i++) {
-      if (ptr_len_cmp(host, host_len, stat_pages[i].module) == 0) {
+      if ((host_len == sizeof(stat_pages[i].module)) && (memcmp(host, stat_pages[i].module, host_len) == 0)) { 
```

Comment:
Interesting. So the old code never checked the length of the .module string. Which presumably meant it could, in theory, do memcmp() outside that buffer?